[TACHYON-1428] Documented parts of the tachyon-servers module in preparation for the checkstyle rule

HorizonNet opened the pull request:
https://tachyon.atlassian.net/browse/TACHYON-1428

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Tachyon-Pull-Request-Builder/7812/
Test PASSed.

apc999 commented on servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java:
```diff
@@ -166,7 +166,7 @@ public void cleanupSessionTempBlocks(long sessionId, List<Long> tempBlockIds) {
    * @throws IllegalArgumentException when location does not belong to tiered storage
    */
-  public long getAvailableBytes(BlockStoreLocation location) {
+  public long getAvailableBytes(BlockStoreLocation location) throws IllegalArgumentException {
```
my view is `IllegalArgumentException` is a subclass of `java.lang.RuntimeException`, which we don't usually put into method signature. Do you have a particular reason of doing this, just to match the javadoc which declares throwing this exception? (if it is for the latter case,  I suggest to update the javadoc instead)

apc999 commented:
@HorizonNet thanks for the PR. I put a comment inline regarding `IllegalArgumentException`. Otherwise the patch looks good to me. Thanks!

HorizonNet commented:
@apc999 Thanks for the comment. It was very helpful. I updated the PR to include the description for the ```IllegalArgumentException```  in the general documentation for the methods and removed the exception from the signature.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Tachyon-Pull-Request-Builder/7820/
Test FAILed.

HorizonNet commented:
Jenkins, test this please.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Tachyon-Pull-Request-Builder/7823/
Test PASSed.

apc999 commented:
+1 LGTM
